pageserver/src/consumption_metrics.rs

```diff
@@ -300,35 +200,29 @@ pub async fn collect_metrics_task(
 
     // Send metrics.
```

Comment:
Why can't we share everything from here on with `proxy`? Looks quite similar to me.

Reply:
umm, we collect metrics in a different way and they have slightly different format.

all shared code is already moved in lib/consumption_metrics

Reply:
I'm speaking of just the chunking + sending part. Those look the same to my eyes.

I won't die on that hill, though.

Reply:
I like @problame's proposal though. Can we leave a TODO comment? I'm sure somebody will pick this up later.